### ISSUE TITLE
How to use event-assisted forecasting?

### ISSUE THREAD

**Issue by Irelandr** [XAITraffic/XTraffic#11]
I am very interested in your paper. Considering that the main contribution of this paper is the introduction of events to assist prediction, but you did not discuss in detail how to do it, only gave the experimental results, and there is no relevant information in the code library, how do you use the collected event data to assist spatiotemporal prediction?

**Comment by Irelandr** [comment]
At the same time, in your readme, you mentioned "For traffic forecasting: We recommend you use the repository LargeST for traffic forecasting tasks." Does your article and data only focus on event reasoning? Can traffic time series data be replaced at will? I am confused about this.

**Comment by gpxlcj** [comment]
Hi,

Thanks very much for your interest in our work.

1) We appreciate you raising this point—while there has been a stream of research on post-incident traffic forecasting and analysis, none have yet provided a public benchmark combining both structured traffic time-series and incident data. The details of the experiment settings can be found in the appendix of the paper, and you can find the code in this repository, except for the traffic forecasting task (as we mentioned, we use LargeST for that task).

2) Yes, our traffic data can be plugged into any forecasting pipeline. That's why we suggest using LargeST—a strong, scalable traffic forecasting benchmark—for continuous prediction experiments separate from the incident-driven tasks. If you're simply building standard traffic forecasting models, feel free to load XTraffic's time series into LargeST's framework.

**Comment by Irelandr** [comment]
Your paper clearly states that you have conducted additional post-event predictions for traffic forecasts and pointed out that introducing event information can bring benefits. However, the main text, appendix, and code base do not specify the detailed settings of "post-event predictions". I doubt its authenticity?

**Comment by gpxlcj** [comment]
I need to clarify a potential misunderstanding of yours regarding the insights conveyed in Section 4.2 of our paper. 

While integrating event (incident) information into traffic forecasting models is indeed a promising direction, it is not the main insight we aim to highlight in Section 4.2. The key observation from our experimental results is that:

**All baselines perform significantly better in predicting on the general test dataset compared to
the incident test dataset.** You can find this sentence in the results part of Section 4.2.

This indicates that existing models are not well-suited for post-incident scenarios. As such, we believe our dataset and benchmark can facilitate future research efforts focused on improving model performance in post-incident conditions.

-----

In our experimental setup, all baseline models are fine-tuned using only general-case training data. We then evaluate them on both:

The general test set, and a post-incident subset of the test set are chosen from the general test set.

-----

I have checked the appendix and the main paper in our arXiv version. We didn't include the details to construct the post-incident test set. 

Below is the process we followed:

1. For each reported incident in D5 (Monterey), we identify the corresponding freeway.

2. We match the incident with the closest sensor (e.g., Sensor A) on the same freeway based on the Absolute Postmile (Abs PM).

3. The incident timestamp is converted into a 5-minute time slot (e.g., 19:13 becomes 19:10–19:15), and the next slot (e.g., 19:15–19:20) is defined as the starting point of post-incident influence.

4. Given the multistep nature of our forecasting task, we select the traffic from 19:15 to 19:45 at Sensor A as a single post-incident test case. (Using traffic from 18:15 to 19:15 as input feature)

While this definition is not perfect, it is still a reasonable starting point and leaves room for refinement. Future work can explore more ways to define and identify post-incident impact windows.
